### PR TITLE
Fix create new file telemetry

### LINKED ISSUE #4781: ODSP: Errors on file creation do not propagate correlation ID / cancel event is not an error
Based on [this investigation](https://teams.microsoft.com/l/message/19:53328301da384b33bcc81111124e3ab4@thread.skype/1610391437693?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=422665a0-7ad3-4d0d-9171-e8881d0397d9&parentMessageId=1610391437693&teamName=FFX%20(Fluid%20Framework%20and%20Experiences)&channelName=Bugs%20%F0%9F%90%9B&createdTime=1610391437693)

fluid:telemetry:OdspDriver:createNewFile_cancel contains sharepoint correlation ID.
But when this error is propagated to fluid:telemetry:Container:ContainerClose, correlation ID is not there.
It would be great for errors to keep all of their properties, including correlation IDs.

I have not looked at details, i.e. what's the issue here.

Also cancel event should be an error event - we already do so in other places (I believe in summarization and loading from snapshot). Not sure if it's one-by-one fixing (like in this place), or logger should just make all cancel event (by default) errors.

### PR DESCRIPTION
**Pull request by jatgarg**
Fixes: https://github.com/microsoft/FluidFramework/issues/4781

Allow reporting cancel event as errors.
Add headers in create new file telemetry.

### CODE REVIEW

File: packages/drivers/odsp-driver/src/createFile.ts [hunk 1]
```diff
@@ -90,9 +90,11 @@ export async function createNewFluidFile(
                 }
                 event.end({
                     headers: Object.keys(headers).length !== 0 ? true : undefined,
+                    ...fetchResponse.commonSpoHeaders,
                 });
                 return content.itemId;
-            });
+            },
+            true);
```
Comment (vladsud) [inline packages/drivers/odsp-driver/src/createFile.ts:97]: Would appreciate if you wait for and merge with https://github.com/microsoft/FluidFramework/pull/5363

**Comment by vladsud** [comment]
Generic question around creation: Should we bake retry logic, similar to how we do with storage / ops in Container layer?
I know that Teams does its own retry loop, and I'd assume other clients do the same. But they are missing on convention of what is retryable and what is not, which is for most part internal to Fluid. So they will retry on something that is not retryable.
At the same time, they may simply give after N retries and that might be good enough?
Any chance you can follow up here, ask around, and form an opinion if work is needed at our layer? I can help with names if needed.

**Review by vladsud — APPROVED** [review]

File: packages/drivers/odsp-driver/src/createFile.ts (hunk 1 above)
Comment (jatgarg) [reply packages/drivers/odsp-driver/src/createFile.ts:97]: sure

**Comment by jatgarg** [comment]
> Generic question around creation: Should we bake retry logic, similar to how we do with storage / ops in Container layer?
> I know that Teams does its own retry loop, and I'd assume other clients do the same. But they are missing on convention of what is retryable and what is not, which is for most part internal to Fluid. So they will retry on something that is not retryable.
> At the same time, they may simply give after N retries and that might be good enough?
> Any chance you can follow up here, ask around, and form an opinion if work is needed at our layer? I can help with names if needed.

I think we should bake the retry logic in loader layer itself based on configurable policies which can be set by hosts/apps. I asked ruchika and some one from onenote and they were of the same opinion. I will follow up on that.

**Comment by vladsud** [comment]
Super - thanks! Note that we if we control retry logic, then our approach to logging would change as well - we would likely not want to log retryable errors as errors (such that we do not spam system while we are offline). Basically we would follow same pattern as everywhere else :)

**Pull request merged**